cocotb/_py_compat.py

```diff
@@ -0,0 +1,109 @@
+"""
```

Comment:
Please add a license header.

Reply:
Since this is a new file, please make it "Copyright cocotb contributors" in the first line (no year needed). I hope we can get these headers cleaned up one day (#804).

Reply:
Updated